go-controller/pkg/ovn/gress_policy.go

```diff
@@ -150,13 +159,29 @@ func (gp *gressPolicy) addPeerPods(pods ...*v1.Pod) error {
 	if config.IPv4Mode && config.IPv6Mode {
 		podIPFactor = 2
 	}
+
+	tmpIpToPod := make(map[string]apitypes.UID)
 	ips := make([]net.IP, 0, len(pods)*podIPFactor)
 	for _, pod := range pods {
 		podIPs, err := util.GetPodIPsOfNetwork(pod, &util.DefaultNetInfo{})
 		if err != nil {
 			return err
 		}
+
 		ips = append(ips, podIPs...)
+		for _, podIP := range podIPs {
+			tmpIpToPod[podIP.String()] = pod.UID
+		}
+	}
+
+	// Sorting to be sure the same lock sequence is done also in `deletePeerPod`
+	sort.Slice(ips, func(i, j int) bool { return ips[i].String() < ips[j].String() })
```

Comment:
this doesn't save us from the duplicate ips, because we iterate based on the ips list, but if we do this based on the `tmpIpToPod` keys, then we can be sure every ip will only be handled once